src/globus_sdk/experimental/globus_app/authorizer_factory.py

```diff
@@ -103,6 +117,16 @@ def store_token_response_and_clear_cache(
         super().store_token_response_and_clear_cache(token_res)
         self._cached_authorizer_expiration = {}
 
+    def clear_cache(self, *resource_servers: str) -> None:
+        if not resource_servers:
+            self._cached_authorizer_expiration = {}
+        else:
+            for resource_server in resource_servers:
+                if resource_server in self._cached_authorizer_expiration:
+                    del self._cached_authorizer_expiration[resource_server]
+
+        super().clear_cache(*resource_servers)
```

Comment:
I see this matching the superclass structure, and I can't help but think that there's some abstraction or interface here we could add to merge this.
What about this?
```python
    if not resource_servers:
        self._cached_authorizer_expiration = {}
    super().clear_cache(*resource_servers)

...

def _clear_resource_server_from_cache(self, resource_server: str) -> None:
    self._cached_authorizer_expiration.pop(resource_server, None)
    super()._clear_resource_server_from_cache(resource_server)
```

and the base class uses that method. Not worth it?

Reply:
We could, I think it's slightly over-engineered to genericize that touch point though & would prefer to keep it the way it is for now.